### ISSUE TITLE
connectivity_measure.fit_transform

### ISSUE THREAD

**Issue by grjd** [nilearn/nilearn#1460]
Hi 
Not sure how to interpret this result when calling fit_transform(timeseries):

1.  I extract the time series for the DMN 
dim_coords = {label[0]:(0, -52, 18),label[1]:(-46, -68, 32), label[2]:(46, -68, 32),label[3]:(1, 50, -5)}
I got 0 correlation matrix
[[ 1.  0. -0. -0.]
 [ 0.  1. -0. -0.]
 [-0. -0.  1. -0.]
 [-0. -0. -0.  1.]]

2. If I change ONLY the second seed, which is now equal to the first seed, all the rest equal
dim_coords = {label[0]:(0, -52, 18),label[1]:(0, -52, 18), label[2]:(46, -68, 32),label[3]:(1, 50, -5)} 
I get this correlation matrix, how can be this possible? 

[[ 1.          0.12200073 -0.01801971  0.12200073]
 [ 0.12200073  1.         -0.06792058  0.91561723]
 [-0.01801971 -0.06792058  1.         -0.06792058]
 [ 0.12200073  0.91561723 -0.06792058  1.        ]]

**Comment by mrahim** [comment]
Hi @grjd , could you plot the timeseries corresponding to dim_coords ?

**Comment by grjd** [comment]
Hi yes I plot the time series no problem

**Comment by grjd** [comment]
![figure_12](https://cloud.githubusercontent.com/assets/7142903/26627880/44c3e758-45fc-11e7-8f67-444ec498e7b4.png)
![figure_6](https://cloud.githubusercontent.com/assets/7142903/26627881/44c61852-45fc-11e7-96bc-2e285417bddc.png)

**Comment by GaelVaroquaux** [comment]
You are putting the second seed equal to the first seed? In this case, you are creating a terribly ill-conditioned system. Estimating the population covariance is now a very hard problem (multi-collinearity).

Behind the hood, the ConnectivityMeasure is using a covariance estimator that detects this and regularizes (the LedoitWolf estimator). What it is saying is that those two time series look too similar, so it doesn't trust their raw correlation, neither how they correlate to the rest.

You could override this and use the maximum likelihood estimate, which would give you answers that match your expectation, but from a statistical estimation standpoint, it's not a good idea.

**Comment by grjd** [comment]
Yes, forget the example with two seeds identical, but my point is that for the second image, 4 perfectly fine time series, the correlation matrix is 0!

I will rephrase the problem: I have 4 seeds s={s1,s2,s3, s4}  get the time series and  connectivity_measure.fit_transform(ts) returns the identity matrix (no correlation between pairs of seeds found)

But, If I change one seed, let us say, s2 for s5, now the connectivity_measure.fit_transform(ts)  for {s1,s5,s3, s4} returns a matrix with correlation >0 between pairs. How can be possible that the correlation between, for example, s1 and s4 is 0 in the first case and and .3 in the second case??

**Comment by GaelVaroquaux** [comment]
That's indeed strange. What are the arguments that you gave when creating the ConnectivityMeasure object?

**Comment by grjd** [comment]
kind_of_corr = 'correlation' #tried also with partial_correlation also return identity matrix
connectivity_measure = ConnectivityMeasure(kind=kind_of_corr)   
#(Attached) time_series.shape == (116, 4)
[time_series.txt](https://github.com/nilearn/nilearn/files/1041210/time_series.txt)

correlation_matrix = connectivity_measure.fit_transform([ts])[0]

**Comment by mrahim** [comment]
It looks like the LedoitWolf estimator shrinks too much towards the identity.
I've found the shrinkage = 1.0

```python
import numpy as np
from sklearn.covariance import LedoitWolf, EmpiricalCovariance

X = np.loadtxt('/tmp/timeseries.txt')
cov = EmpiricalCovariance(assume_centered=True)
cov.fit(X)
print(cov.covariance_)

lw = LedoitWolf(assume_centered=True)
lw.fit(X)
print(lw.covariance_)
print('shrinkage = %.4f' % lw.shrinkage_)
```
```
[[ 1.          0.13324425 -0.0196804  -0.11378572]
 [ 0.13324425  1.         -0.0741801  -0.01235692]
 [-0.0196804  -0.0741801   1.         -0.17500451]
 [-0.11378572 -0.01235692 -0.17500451  1.        ]]
[[ 1.  0. -0. -0.]
 [ 0.  1. -0. -0.]
 [-0. -0.  1. -0.]
 [-0. -0. -0.  1.]]
shrinkage = 1.0000
```
[timeseries.txt](https://github.com/nilearn/nilearn/files/1041336/timeseries.txt)

**Comment by grjd** [comment]
Thank you! 
I am not sure how to interpret the shrinkage for the Leit covariance estimator. What does it say about the time series covariance? I wonder if there is a way to compute the optimal
shrinkage intensity

**Comment by GaelVaroquaux** [comment]
> It looks like the LedoitWolf estimator shrinks too much towards the identity.
> I've found the shrinkage = 1.0

Indeed, I now remember that we hit this a while ago. It surprised a lot
back then, and when I checked, I found that it was correct accordingly to
the Ledoit-Wolf model. Maybe this reveals an imperfection of the
LedoitWolf model, but that's what we have for now.

@grjd: if you really want raw correlations, you can disable the
LedoitWolf estimate as following:

<pre>
from sklearn import covariance
connectivity_measure = connectivity_measure(kind='correlation', 
                 cov_estimator=covariance.EmpiricalCovariance())
</pre>

With regards to what the results that you are observing mean with
LedoitWolf: they mean that the LedoitWolf estimator finds the data too
noisy and the time series too short to estimate reliably connectivity.

We're working on better estimates that should perform better (@mrahim:
this should be a good example for your paper). There is a paper that's
accepted, but we need much more validation before we can consider
integrating this in nilearn.

**Comment by bthirion** [comment]
Nothing left to do. Closing.
